src/EPR.Calculator.Service.Common/AzureSynapse/AzureSynapseRunner.cs

```diff
@@ -1,137 +1,156 @@
+
+                    this.logger.LogInformation($"pipelineStatus for pipelineRunId {pipelineRunId}: {pipelineStatus}");
+                    if (pipelineStatus.Contains("Succeeded"))
                     {
                         break;
                     }
                 }
-                catch
+                catch (Exception ex)
                 {
-                    // Something went wrong retrieving the status,but we're going to try again,
+                    this.logger.LogError($"Error occurred in Azure Synapse Runner: {ex.StackTrace}");
+                    this.logger.LogInformation($"Check count inside exception: {checkCount}");
+
+                    // Something went wrong retrieving the status, but we're going to try again,
                     // so ignore it unless this is the last try.
-                    if (checkCount >= args.MaxChecks)
+                    if (checkCount >= maxCheckCount)
                     {
                         break;
                     }
                 }
 
-                await Task.Delay(args.CheckInterval);
-            }
-            while (checkCount < args.MaxChecks);
+                var checkInterval = args.CheckInterval;
+
+                this.logger.LogInformation($"CheckInterval: {checkInterval}");
+                this.logger.LogInformation($"Task started at: {DateTime.Now}");
 
-            // Record success/failure to the database using the web API.
-            using var client = this.PipelineClientFactory.GetStatusUpdateClient(args.StatusUpdateEndpoint);
-            var statusUpdateResponse = await client.PostAsync(
-                    args.StatusUpdateEndpoint.ToString(),
-                    GetStatusUpdateMessage(args.CalculatorRunId, pipelineStatus == nameof(PipelineStatus.Succeeded)));
+                await Task.Delay(TimeSpan.FromMilliseconds(checkInterval));
+                this.logger.LogInformation($"Task completed at: {DateTime.Now} with CheckInterval {checkInterval} and checkCount is {checkCount}");
+            }
+            while (checkCount < maxCheckCount);
 
-            #if DEBUG
-            Debug.WriteLine(statusUpdateResponse.Content.ReadAsStringAsync().Result);
-            #endif
+            this.logger.LogInformation($"Azure Synapse Runner completed at: {DateTime.Now} and checkCount is {checkCount}");
 
-            return pipelineStatus == nameof(PipelineStatus.Succeeded) && statusUpdateResponse.IsSuccessStatusCode;
+            return pipelineStatus == nameof(PipelineStatus.Succeeded);
         }
 
         /// <summary>
-        /// Build the JSON content of the status update API call.
+        /// Starts a pipeline run.
         /// </summary>
-        private static StringContent GetStatusUpdateMessage(int calculatorRunId, bool pipelineSucceeded)
-            => new StringContent(
-                JsonSerializer.Serialize(new
-                {
-                    runId = calculatorRunId,
-                    updatedBy = "string",
```

Comment:
Updated by can you replace with a parameter value like "Azure Function" which can be changed later

Reply:
Resolved in PR#16